File: fredriv/day01/calories.scala
import scala.io.Source

@main def calories =
  val data = Source.fromFile("input.txt").getLines.mkString("\n")
  val caloriesPerElf =
    for calories <- data.split("\n\n")
    yield calories.split("\n").map(_.toInt).sum

  println(caloriesPerElf.max)
  println(caloriesPerElf.sorted(Ordering.Int.reverse).take(3).sum)

File: archive/2022/larssnek/1/calories.py
from collections import defaultdict

i = 1
elves = defaultdict(lambda: 0)

with open("input.txt", "r") as f:
    while line := f.readline():
        if line != '\n':
            elves[i] += int(line)
        else:
            i +=1

print(max(elves.values()))
print(sum(sorted(elves.values())[-3:]))
